sympy/matrices/matrices.py

```diff
@@ -78,547 +71,65 @@ def __repr__(self):
-        algorithm which is an extension of the well known Gaussian
-        elimination method. This approach is best suited for dense
-        symbolic matrices and will result in a determinant with
-        minimal number of fractions. It means that less term
-        rewriting is needed on resulting formulae.
-
-        Parameters
-        ==========
-
-        dotprodsimp : bool, optional
-            Specifies whether intermediate term algebraic simplification is used
-            during matrix multiplications to control expression blowup and thus
-            speed up calculation.
-
-        TODO: Implement algorithm for sparse matrices (SFF),
-        http://www.eecis.udel.edu/~saunders/papers/sffge/it5.ps.
-        """
-
-        # Recursively implemented Bareiss' algorithm as per Deanna Richelle Leggett's
-        # thesis http://www.math.usm.edu/perry/Research/Thesis_DRL.pdf
-        def bareiss(mat, cumm=1):
-            if mat.rows == 0:
-                return mat.one
-            elif mat.rows == 1:
-                return mat[0, 0]
-
-            # find a pivot and extract the remaining matrix
-            # With the default iszerofunc, _find_reasonable_pivot slows down
-            # the computation by the factor of 2.5 in one test.
-            # Relevant issues: #10279 and #13877.
-            pivot_pos, pivot_val, _, _ = _find_reasonable_pivot(mat[:, 0],
-                                         iszerofunc=iszerofunc)
-            if pivot_pos is None:
-                return mat.zero
-
-            # if we have a valid pivot, we'll do a "row swap", so keep the
-            # sign of the det
-            sign = (-1) ** (pivot_pos % 2)
-
-            # we want every row but the pivot row and every column
-            rows = list(i for i in range(mat.rows) if i != pivot_pos)
-            cols = list(range(mat.cols))
-            tmp_mat = mat.extract(rows, cols)
-
-            def entry(i, j):
-                ret = (pivot_val*tmp_mat[i, j + 1] - mat[pivot_pos, j + 1]*tmp_mat[i, 0]) / cumm
-                if dotprodsimp:
-                    return _dotprodsimp(ret)
-                elif not ret.is_Atom:
-                    return cancel(ret)
-                return ret
-
-            return sign*bareiss(self._new(mat.rows - 1, mat.cols - 1, entry), pivot_val)
-
-        return bareiss(self)
-
-    def _eval_det_berkowitz(self, dotprodsimp=None):
-        """ Use the Berkowitz algorithm to compute the determinant.
-
-        Parameters
-        ==========
-
-        dotprodsimp : bool, optional
-            Specifies whether intermediate term algebraic simplification is used
-            during matrix multiplications to control expression blowup and thus
-            speed up calculation.
-        """
+    def _find_reasonable_pivot(self, iszerofunc=_iszero, simpfunc=_simplify):
+        return _find_reasonable_pivot(self, iszerofunc=iszerofunc,
+                simpfunc=iszerofunc)
```

Comment:
Currently Matrix does not have a `_find_reasonable_pivot` method so I don't think we need to add one.

Reply:
Like I said, @sylee asked for this and I can see the reasoning if sparse matrices or numpy matrices can implement this better in their cases but if you want it gone its gone.

Reply:
Oh I forgot